test/test612.js

```diff
@@ -0,0 +1,144 @@
+if(typeof exports === 'object') {
+	var assert = require("assert");
+	var alasql = require('..');
+	var fs = require("fs");
+}
+
+var test = '612'; // insert test file number
+
+describe('Test '+test+' - INTO CSV', function() {
+	
+	before(function(){
+		alasql('CREATE DATABASE test' + test);
+		alasql('USE test' + test);
+		alasql('CREATE TABLE one (a INT, b VARCHAR)');
+		alasql('INSERT INTO one VALUES (10, \'swoll\'),' +
+                        '(11, \'muscles\')');
+
+	});
+
+	after(function(){
+		alasql('DROP DATABASE test' + test);
+		fs.unlink('test612-0.csv', function(err) {  });
+		fs.unlink('test612-1.csv', function(err) {  });
+		fs.unlink('test612-2.csv', function(err) {  });
+		fs.unlink('test612-3.csv', function(err) {  });
+        });
+
+	it('With quote = \'\', single string value', function(){
+		alasql("SELECT 'swing' AS `colname` INTO CSV('test612-0', {quote:''})");
+		fs.open('test612-0.csv', 'r', function(err, fd) {
+			if (err) {
+				console.error(err);
+			}
+
+			var buf = new Buffer(1024);
+			var csvstr = '';
+			fs.read(fd, buf, 0, buf.length, 0, function(err, bytes) {
+				if (err) {
+					console.error(err);
+				}
+
+				if (bytes > 0) {
+					csvstr = buf.slice(0, bytes).toString();
+				}
+
+				assert(csvstr === 'colname\n"swing"');
+		
+				fs.close(fd, function(err) {
+					if (err) {
+						console.error(err);
+					}
+				});
+			});
+        	});
+	});
+
+	it('With quote = \'\', single multiword string value', function(){
+		alasql("SELECT 'swing out' AS `colname` INTO CSV('test612-1', {quote:''})");
+		fs.open('test612-1.csv', 'r', function(err, fd) {

```

Comment:
Thank you for that.  I've made the changes, but I'm having issues with the assertions.  Here's what the first `it()` call looks like for me now:

```javascript
it('With quote = \'\', single string value', function(){
		return alasql.promise("SELECT 'swing' AS `colname` INTO CSV('test612-0', {quote:''})")
			.then(function(state) {
				console.log(fs.readFileSync('test612-0.csv', 'utf8') + '\n\n');
				assert(fs.readFileSync('test612-0.csv', 'utf8') === 'colname\nswing\n');
		}, function(err) {});
	});
```
It appears that the string `readFileSync()` is pulling is not not what I imagine it to be because the assertion fails.  I've also tried the following strings and various other combinations I've forgotten:
`'colname\nswing\n\ufeff'`
`'colname\r\nswing\r\n\ufeff'`
`'colname\nswing\n\n`

I'm at a loss for how to write the assertion. Any ideas on where I might be going wrong with this?

Also, I chose the return the Promise rather than use `done()` because I read that mocha will catch the errors automatically.  That part seems to be working now, whereas I was getting errors using `done()` as you advised.